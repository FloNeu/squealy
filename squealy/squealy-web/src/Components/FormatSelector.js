import React, {Component} from 'react'
import { RESPONSE_FORMATS } from '../Constant'

export default class FormatSelector extends Component {
  render () {
    const { selectedFormat, apiResponseGetter, selectedApiIndex} = this.props
    return (
            <div>
            <form>
            {
              Object.keys(RESPONSE_FORMATS).map(function(key){
                return(
                  <label className="radio-inline" key={key}>
                    <input type="radio" 
                      name={'optradio_'+selectedApiIndex}
                      value={RESPONSE_FORMATS[key].value}
                      checked={RESPONSE_FORMATS[key].value===selectedFormat}
                      onClick={(e) =>
                      apiResponseGetter(e, RESPONSE_FORMATS[key].value)} />
                    {RESPONSE_FORMATS[key].displayName}
                  </label>
                )
              })
            }
            </form>
            </div>
    )
  }
}
