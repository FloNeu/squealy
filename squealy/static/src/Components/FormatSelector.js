import React, {Component} from 'react'
import { RESPONSE_FORMATS } from '../Constant'

export default class FormatSelector extends Component {
  render () {
    const { selectedFormat, apiResponseGetter } = this.props
    return (
            <div>
            {
              Object.keys(RESPONSE_FORMATS).map(function(key){
                return(
                  <label className="radio-inline" key={key}>
                    <input type="radio" name="optradio"
                      checked={(selectedFormat===RESPONSE_FORMATS[key].value)}
                      onClick={(e) =>
                      apiResponseGetter(e, RESPONSE_FORMATS[key].value)} />
                    {RESPONSE_FORMATS[key].displayName}
                  </label>
                )
              })
            }
            </div>
    )
  }
}
