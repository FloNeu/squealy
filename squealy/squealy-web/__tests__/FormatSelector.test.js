import React from 'react';
import renderer from 'react-test-renderer';

import FormatSelector from '../src/Components/FormatSelector'

const apiResponseGetter =(event, value) => {
  return value
}

test('It returns the format which is clicked', () => {
   const component = renderer.create(
    <FormatSelector selectedFormat='table'
                apiResponseGetter={apiResponseGetter} />
  );

  let tree = component.toJSON();

  expect(tree.children[0].children[0].props.onClick()).toBe('table')
  expect(tree).toMatchSnapshot();

})
