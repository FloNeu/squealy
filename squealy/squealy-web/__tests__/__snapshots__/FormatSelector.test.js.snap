exports[`test It returns the format which is clicked 1`] = `
<div>
  <label
    className="radio-inline">
    <input
      checked={true}
      name="optradio"
      onClick={[Function onClick]}
      type="radio" />
    Table
  </label>
  <label
    className="radio-inline">
    <input
      checked={false}
      name="optradio"
      onClick={[Function onClick]}
      type="radio" />
    JSON
  </label>
  <label
    className="radio-inline">
    <input
      checked={false}
      name="optradio"
      onClick={[Function onClick]}
      type="radio" />
    Google Charts
  </label>
  <label
    className="radio-inline">
    <input
      checked={false}
      name="optradio"
      onClick={[Function onClick]}
      type="radio" />
    Highchart
  </label>
</div>
`;
